components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -44,39 +56,76 @@ public void init() throws ArtifactSynchronizerException {
+            return gatewayRuntimeArtifact;
+        } catch (IOException e) {
+            String msg = "Error while executing the http client";
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
         }
-        return gatewayRuntimeArtifacts;
     }
 
     @Override
     public List<String> retrieveAllArtifacts(String label) throws ArtifactSynchronizerException {
         List<String> gatewayRuntimeArtifactsArray = new ArrayList<>();
+        CloseableHttpResponse httpResponse = null;
         try {
-            List<ByteArrayInputStream> baip = apiMgtDAO.getAllGatewayPublishedAPIArtifacts(label);
-            for (ByteArrayInputStream byteStream :baip){
-                byte[] bytes = ByteStreams.toByteArray(byteStream);
-                String  gatewayRuntimeArtifacts = new String(bytes);
-                gatewayRuntimeArtifactsArray.add(gatewayRuntimeArtifacts);
+            String endcodedgatewayLabel= URLEncoder.encode(label, APIConstants.DigestAuthConstants.CHARSET);
+            String path = APIConstants.GatewayArtifactSynchronizer.GATEAY_SYNAPSE_ARTIFACTS
+                    + "?gatewayLabel="+ endcodedgatewayLabel;
+            String endpoint = baseURL + path;
+            httpResponse = invokeService(endpoint);
+            String responseString;
+            if (httpResponse.getEntity() != null ) {
+                responseString = EntityUtils.toString(httpResponse.getEntity(),
+                        APIConstants.DigestAuthConstants.CHARSET);
+                httpResponse.close();
+            } else {
+                throw new ArtifactSynchronizerException("HTTP response is empty");
             }
-            if (log.isDebugEnabled()){
-                log.debug("Successfully retrieved Artifacts from DB");
+            JSONObject artifactObject = new JSONObject(responseString);
+            JSONArray jArray = (JSONArray)artifactObject.get("list");
+            if (jArray != null) {
+                for (int i = 0; i < jArray.length(); i++) {
+                    gatewayRuntimeArtifactsArray.add(jArray.get(i).toString());
+                }
             }
-        } catch (APIManagementException | IOException e) {
-            throw new ArtifactSynchronizerException("Error retrieving Artifact from DB", e);
+            return gatewayRuntimeArtifactsArray;
+        } catch (IOException e) {
+            String msg = "Error while executing the http client";
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
         }
-        return gatewayRuntimeArtifactsArray;
+    }
+
+    private CloseableHttpResponse invokeService(String endpoint) throws IOException {
+        HttpGet method = new HttpGet(endpoint);
+        URL url = new URL(endpoint);
+        String username = eventHubConfigurationDto.getUsername();
+        String password = eventHubConfigurationDto.getPassword();
```

Comment:
password must be a char[]

Reply:
Yes . But the` getPassword()` method returns a string (converting `char[]` --> `String`)
https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dto/EventHubConfigurationDto.java#L90

Reply:
we need to change that. @tharindu1st was there any reason not to do that?

Reply:
we need to change it